src/main/resources/fileTemplates/testMeTests/JUnit4 & Powermock.java.ft

```diff
@@ -0,0 +1,94 @@
+#parse("TestMe macros.java")
+#set($hasMocks=$MockitoMockBuilder.hasMocks($TESTED_CLASS))
+#if($PACKAGE_NAME)
+package ${PACKAGE_NAME};
+#end
+
+import org.junit.Assert;
+import org.junit.Test;
+#if($hasMocks)
+import static org.powermock.api.mockito.PowerMockito.*;
+import org.powermock.api.mockito.PowerMockito;
+import org.powermock.core.classloader.annotations.PowerMockIgnore;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
+import org.powermock.reflect.Whitebox;
+import org.junit.Before;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import org.junit.runner.RunWith;
+import static org.mockito.ArgumentMatchers.*;
+import static org.mockito.Mockito.verify;
+#end
+
+#parse("File Header.java")
+@RunWith(PowerMockRunner.class)
+@PrepareForTest({${TESTED_CLASS.name}.class})
+@PowerMockIgnore("javax.management.*")
+public class ${CLASS_NAME} {
+    #renderMockedFields($TESTED_CLASS.fields,$hasMocks)
+    #renderPowerMockTestSubjectInit($TESTED_CLASS,$TestSubjectUtils.hasTestableInstanceMethod($TESTED_CLASS.methods),$hasMocks)
+
+    ## all methods under powermock framework can be tested
+    #foreach($method in $TESTED_CLASS.methods)
```

Comment:
when some complex private method called in public method，maybe we would like to stud the private method, and test the private method alone. for example may be we can stud private `deleteComponentRelated` method in `deleteByNpmName` and test it alone, because `deleteComponentRelated` has some called method of mocked filed like `lowCodeComponentDao`, and for private method `hasNpmName` we need't test it alone because it has no called method from mocked filed .
````java
    @Override
    public RestResponse<Void> deleteByNpmName(String npmName) {
        RestResponse<Void> response = new RestResponse<>();
        List<LowCodeComponentEntity> componentEntityList = lowCodeComponentDao.queryComponentList();
        if (!hasNpmName(npmName, componentEntityList)) {
            response.setCode(RestResultCode.COMPONENT_NOT_EXIST_ERROR.getCode());
            response.setMsg(RestResultCode.COMPONENT_NOT_EXIST_ERROR.getMessage());
            return response;
        }
        return deleteComponentRelated(npmName);
    }

    private RestResponse<Void> deleteComponentRelated(String npmName) {
        List<LowCodeComponentFileSourceEntity> sourceEntityList = lowCodeComponentDao.findComponentFileSource(npmName);
        List<String> fileIdList = new ArrayList<>();
        for (LowCodeComponentFileSourceEntity sourceEntity : sourceEntityList) {
            lowCodeFileSourceDao.deleteFile(sourceEntity.getFileId());
            fileIdList.add(sourceEntity.getFileId());
        }
        lowCodeComponentDao.deleteComponentFileSource(npmName);
        lowCodeFileSourceDao.batchDeleteFileSource(fileIdList);
       return RestResponse.success();
    }

   private boolean hasNpmName(String npmName, List<LowCodeComponentEntity> componentEntityList) {
        if (CollUtil.isNotEmpty(componentEntityList)) {
            for (LowCodeComponentEntity entity : componentEntityList) {
                if (npmName.equals(entity.getNpmName())) {
                    return true;
                }
            }
        }
        return false;
    }
````

what's more maybe we can add a setting to generate all test method including all pivate method, because many real used people they write unit tests solely to meet the coverage quota.

Reply:
These concerns of stubbing/testing internal private methods are typically a result of design issues. Adhering to few simple unit test principals where private methods are not tested - helps in encouraging better design. when private methods are not tested - it's easier to maintain code by refactoring private methods signatures, adding/removing private methods with less impact on public methods/interfaces. So existing UT can still be used for regression testing with fewer adjustments if needed.
BTW, in examples given, a better design would delegate _deleteByNpmName_ method from controller to a *Service class that would implement the logic, call the dao and let the Controller map a DTO result object to _RestResponse_.
Stubbing/testing internal private method is a compromise sometimes done in legacy code, code that cannot be refactored for some reason, or (sorry for saying so) due to design mistakes where developers choose to introduce bad practices in UT instead of fixing/refactoring design of tested code. So it's an atypical use case and should not be the default behavior. surely not generate such stubs/tests for *all* private methods.
An option to test/stub private methods should be offered with custom settings when generating a test, or as an [intention](https://plugins.jetbrains.com/docs/intellij/code-intentions.html) that adds such method (could be more complicated to get the relevant context). as short term solution - users can setup a Custom Template. If you'd like, you can create a Gist with such template code - I can link and reference that Gist from TestMe docs.